https://bugs.freedesktop.org/show_bug.cgi?id=24088
http://bugs.gentoo.org/show_bug.cgi?id=286005
http://bugs.gentoo.org/show_bug.cgi?id=288051

From 6975c54a5cd4f1c6d91401a9cfc90d618f5516ee Mon Sep 17 00:00:00 2001
From: Heiko Przybyl <zuxez@cs.tu-berlin.de>
Date: Tue, 22 Sep 2009 20:07:48 +0200
Subject: [PATCH 1/1] Avoid usage of get_current_dir_name() on non-GNU targets.

The testsuite uses get_current_dir_name() to determine the current
directory for the bootstrap download of the images. Unfortunately this
function is a GNU-extension and thus not available on platforms that
aren't GNU, e.g. MacOSX. So guard get_current_dir_name() with _GNU_SOURCE
and if not available fall back to POSIX getcwd().
---
 testsuite/testsuite.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/testsuite/testsuite.cpp b/testsuite/testsuite.cpp
index 402f7ba..9e7d8da 100644
--- a/testsuite/testsuite.cpp
+++ b/testsuite/testsuite.cpp
@@ -853,7 +853,12 @@ int main(int argc, char ** argv)
             break;
         case 'd':
             if(optarg[0] != '/') {
+#ifdef _GNU_SOURCE
                 char * dir = get_current_dir_name();
+#else
+                char * dir = (char *) malloc(PATH_MAX * sizeof(char));
+                getcwd(dir, PATH_MAX);
+#endif
                 download_dir = dir;
                 download_dir += '/';
                 download_dir += optarg;
-- 
1.6.4.2

